polynote-spark-runtime/src/main/scala/polynote/runtime/spark/reprs/SparkReprsOf.scala

```diff
@@ -203,5 +203,41 @@ object SparkReprsOf {
     }
   }
 
+  implicit val sparkSession: SparkReprsOf[SparkSession] = {
+    instance {
+      sess =>
+        val uiLink = sess.sparkContext.uiWebUrl.map { url =>
+          s"""<span class="field-name">Spark UI</span><a href="$url" class="link" target="_blank">$url</a>"""
+        }.getOrElse("""<span class="field-name error">Spark UI url not found!</span>""")
+
+        // TODO: this is pretty fragile: tight coupling to the display_content implementation.
+        //  We should have a better way to inject data for display instead of doing this.
```

Comment:
One alternative might be instead of raw HTML, we can create some sort of display repr that will get properly filled in by the frontend code rather than having to duplicate it here.

Reply:
TBH I was picturing the spark session display as just a simple HTML table.